apps/shared/avifpng.c

```diff
@@ -312,6 +302,89 @@ avifBool avifPNGRead(const char * inputFilename,
+        if (png_get_iCCP(png, info, &iccpProfileName, &iccpCompression, &iccpData, &iccpDataLen) == PNG_INFO_iCCP) {
+            avifImageSetProfileICC(avif, iccpData, iccpDataLen);
+        } else if (png_get_sRGB(png, info, &srgbIntent) == PNG_INFO_sRGB) {
+            // srgbIntent ignored
+            avif->colorPrimaries = AVIF_COLOR_PRIMARIES_BT709;
+            avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
+        } else {
+            avifBool needToGenerateICC = AVIF_FALSE;
+            double gamma;
+            double wX, wY, rX, rY, gX, gY, bX, bY;
+            float primaries[8];
+            if (png_get_gAMA(png, info, &gamma) == PNG_INFO_gAMA) {
+                gamma = 1.0 / gamma;
+                avif->transferCharacteristics = avifTransferCharacteristicsFindWithGamma((float)gamma);
+                if (avif->transferCharacteristics == AVIF_TRANSFER_CHARACTERISTICS_UNKNOWN) {
+                    needToGenerateICC = AVIF_TRUE;
+                }
+            } else {
+                // No gamma information in file. Assume the default value.
+                // PNG specification (Third Edition) Section 13.13:
+                // A display exponent of 2.2 should be used unless detailed calibration measurements
+                // are available for the particular CRT used.
+                gamma = 2.2;
+            }
+
+            if (png_get_cHRM(png, info, &wX, &wY, &rX, &rY, &gX, &gY, &bX, &bY) == PNG_INFO_cHRM) {
+                primaries[0] = (float)rX;
+                primaries[1] = (float)rY;
+                primaries[2] = (float)gX;
+                primaries[3] = (float)gY;
+                primaries[4] = (float)bX;
+                primaries[5] = (float)bY;
+                primaries[6] = (float)wX;
+                primaries[7] = (float)wY;
+                avif->colorPrimaries = avifColorPrimariesFind(primaries, NULL);
+                if (avif->colorPrimaries == AVIF_COLOR_PRIMARIES_UNKNOWN) {
+                    needToGenerateICC = AVIF_TRUE;
+                }
+            } else {
+                // No chromaticity information in file. Assume the default value.
+                // PNG specification (Third Edition) Section 13.14:
+                // PNG decoders may wish to do this for PNG datastreams with no cHRM chunk.
+                // In this case, a reasonable default would be the CCIR 709 primaries [ITU-R_BT.709].
+                avifColorPrimariesGetValues(AVIF_COLOR_PRIMARIES_BT709, primaries);
+            }
+
+            if (needToGenerateICC) {
+                avif->colorPrimaries = AVIF_COLOR_PRIMARIES_UNSPECIFIED;
+                avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED;
+                fprintf(stderr,
+                        "INFO: legacy color space information not matching any CICP value found in file %s. libavif is generating an ICC profile for it."
```

Comment:
They're not deprecated as far as I know. Even calling them "legacy" is a bit of a stretch. Maybe just "png color space information"?

Reply:
Let's take part of Maryla's suggestion: add "PNG" before "color space information". I also suggest reorder the words in line 411 as follows:

```
"INFO: legacy PNG color space information found in file %s not matching any CICP value. libavif is generating an ICC profile for it."
```

I think we can also just say "the PNG gAMA and cHRM chunks" instead of "legacy PNG color space information".

Reply:
Done.
I personally don't want to tell these weirdly capitalized chunk names (through I know it's meaningful) to users. It does not help to understand the situation, unless the user is familiar with PNG specification and color management.